tests/test_cliCmdQueryCollections.py

```diff
@@ -56,41 +75,159 @@ def test_all(self):
 
             # Verify collections that were created are found by
             # query-collections.
-            result = runner.invoke(cli, ["query-collections", "here"])
-            self.assertEqual({"collections": [run, tag]}, yaml.safe_load(result.output))
+            result = self.runner.invoke(cli, ["query-collections", "here"])
+            self.assertEqual(result.exit_code, 0, clickResultMsg(result))
+            expected = Table((("ingest/run", "tag"), ("RUN", "TAGGED")),
+                             names=("Name", "Type"))
+            self.assertAstropyTablesEqual(readTable(result.output), expected)
 
             # Verify that with a glob argument, that only collections whose
             # name matches with the specified pattern are returned.
-            result = runner.invoke(cli, ["query-collections", "here", "t*"])
-            self.assertEqual({"collections": [tag]}, yaml.safe_load(result.output))
+            result = self.runner.invoke(cli, ["query-collections", "here", "t*"])
+            self.assertEqual(result.exit_code, 0, clickResultMsg(result))
+            expected = Table((("tag",), ("TAGGED",)),
+                             names=("Name", "Type"))
+            self.assertAstropyTablesEqual(readTable(result.output), expected)
 
             # Verify that with a collection type argument, only collections of
             # that type are returned.
-            result = runner.invoke(cli, ["query-collections", "here", "--collection-type", "RUN"])
-            self.assertEqual({"collections": [run]}, yaml.safe_load(result.output))
+            result = self.runner.invoke(cli, ["query-collections", "here", "--collection-type", "RUN"])
+            self.assertEqual(result.exit_code, 0, clickResultMsg(result))
+            expected = Table((("ingest/run",), ("RUN",)),
+                             names=("Name", "Type"))
+            self.assertAstropyTablesEqual(readTable(result.output), expected)
+
+
+class ChainedCollectionsTest(ButlerTestHelper, unittest.TestCase):
+
+    @staticmethod
+    def _mock_export(refs: Iterable[DatasetRef], *,
+                     directory: Optional[str] = None,
+                     transfer: Optional[str] = None) -> Iterable[FileDataset]:
+        """A mock of `Datastore.export` that satisfies the requirement that
+        the refs passed in are included in the `FileDataset` objects
+        returned.
+
+        This can be used to construct a `Datastore` mock that can be used
+        in repository export via::
+
+            datastore = unittest.mock.Mock(spec=Datastore)
+            datastore.export = _mock_export
+
+        """
+        for ref in refs:
+            yield FileDataset(refs=[ref],
+                              path="mock/path",
+                              formatter="lsst.daf.butler.formatters.json.JsonFormatter")
+
+    @staticmethod
+    def _mock_get(ref: DatasetRef, parameters: Optional[Mapping[str, Any]] = None
```

Comment:
Rather than copying this from test_simpleButler.py, please put the mocks in a DatastoreMock base class in the tests hierarchy.